packages/loopring_v3/contracts/amm/libamm/AmmData.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2017 Loopring Technology Limited.
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "../../core/iface/ExchangeData.sol";
+import "../../core/iface/IExchangeV3.sol";
+
+
+/// @title AmmData
+library AmmData
+{
+    function LP_TOKEN_BASE() internal pure returns (uint) { return 10 ** 18; }
+    function LP_TOKEN_INITIAL_SUPPLY() internal pure returns (uint) { return 100 * LP_TOKEN_BASE(); }
+    function MAX_AGE_REQUEST_UNTIL_POOL_SHUTDOWN() internal pure returns (uint) { return 7 days; }
+    function MIN_TIME_TO_UNLOCK() internal pure returns (uint) { return 1 days; }
+
+    enum PoolTransactionType
+    {
+        NOOP,
+        JOIN,
+        EXIT
+    }
+
+    struct PoolConfig
+    {
+        address   exchange;
+        string    poolName;
+        uint32    accountID;
+        address[] tokens;
+        uint96[]  weights;
+        uint8     feeBips;
+        string    tokenSymbol;
+    }
+
+    struct PoolJoin
+    {
+        address  owner;
+        bool     fromLayer2;
+        uint     minPoolAmountOut;
+        uint96[] maxAmountsIn;
+        uint32[] storageIDs;
+        uint     validUntil;
+        uint96   fee;
```

Comment:
Seems like we should have an array of fees here, otherwise there's a fixed fee amount for all different tokens?